modules/AlphaWalletFoundation/AlphaWalletFoundation/RPC/BlockchainProvider.swift

```diff
@@ -234,3 +236,20 @@ extension JSONRPCKit.JSONRPCError: LocalizedError {
         }
     }
 }
+
+public enum GasPriceSurchargeding {
+    case percentage(BigUInt)
+    case fixed(BigUInt)
+
+    public func surchargedGasPrice(estimatedGasPrice: BigUInt) -> (value: BigUInt, surchargeding: BigUInt) {
+        let surchargeding: BigUInt
+        switch self {
+        case .percentage(let gasPriceSurchargePercent):
+            surchargeding = BigUInt(Int(Double(estimatedGasPrice) / 100.0 * Double(gasPriceSurchargePercent)))
```

Comment:
Is it better like this? Less loss of precision and having to think about conversions when reading the code:

```
surchageding = estimatedGasPrice * gasPriceSurchargePercent / BigUInt(100)
```